From d13acbcb4f9da501a3b9164d2f6a3d88b26f5a32 Mon Sep 17 00:00:00 2001
From: Martin Jansa <Martin.Jansa@gmail.com>
Date: Fri, 31 Jan 2020 01:10:20 +0100
Subject: [PATCH] qwaylandwindow_p.h: update for changed QPlatformWindow API

* startSystemMove doesn't have any parameters since qtbase:
  commit a611c632bb906cf77dd3af29ddd7b166f79ad1b0
  Author: Johan Klokkhammer Helsing <johan.helsing@qt.io>
  Date:   Tue Feb 6 17:21:13 2018 +0100

    Add QWindow::startSystemMove and startSystemResize

    This can be used to create custom client side window decorations.

    Refactors the xcb implementation to use edges instead of corners and we now use
    the last mouse position for `root_x` and `root_y` in the `_NET_WM_MOVERESIZE`
    event. Touch has also been changed, so just pick a point that's currently being
    pressed.

    The workaround for QTBUG-69716 has now been moved to QSizeGrip, as the comment
    in the bug report says that it should ideally be fixed at the widget level.

    On Windows, we no longer abort when GetSystemMenu returns false. I assume this
    code was added to check whether the window didn't have any decorations and not
    resize in that case. However, since the point of this patch is to let windows
    without native decorations resize/move, it makes most sense to remove the
    check.

    Adds a manual test, which calls QWindow::startSystemMove and startSystemResize
    on touch and mouse events.

    [ChangeLog][QtGui] Added API for starting interactive window resize and move
    operations handled by the system.

Signed-off-by: Martin Jansa <Martin.Jansa@gmail.com>

Change-Id: Ieb9bede68e5a8c82f4f8fac4456852c3defe828e
Signed-off-by: Martin Jansa <Martin.Jansa@gmail.com>
---
 src/client/qwaylandwindow.cpp | 3 +--
 src/client/qwaylandwindow_p.h | 2 +-
 2 files changed, 2 insertions(+), 3 deletions(-)

diff --git a/src/client/qwaylandwindow.cpp b/src/client/qwaylandwindow.cpp
index 0a623e9a..a54d8c9f 100644
--- a/src/client/qwaylandwindow.cpp
+++ b/src/client/qwaylandwindow.cpp
@@ -1181,9 +1181,8 @@ void QWaylandWindow::propagateSizeHints()
         mShellSurface->propagateSizeHints();
 }
 
-bool QtWaylandClient::QWaylandWindow::startSystemMove(const QPoint &pos)
+bool QtWaylandClient::QWaylandWindow::startSystemMove()
 {
-    Q_UNUSED(pos);
     if (auto seat = display()->lastInputDevice())
         return mShellSurface && mShellSurface->move(seat);
     return false;
diff --git a/src/client/qwaylandwindow_p.h b/src/client/qwaylandwindow_p.h
index 35fa1fdf..8a22bacc 100644
--- a/src/client/qwaylandwindow_p.h
+++ b/src/client/qwaylandwindow_p.h
@@ -196,7 +196,7 @@ public:
     void propagateSizeHints() override;
     void addAttachOffset(const QPoint point);
 
-    bool startSystemMove(const QPoint &pos) override;
+    bool startSystemMove() override;
 
     void timerEvent(QTimerEvent *event) override;
     void requestUpdate() override;
